lib/Arduino_Core_A3ides/cores/arduino/HardwareSerial.cpp

```diff
@@ -15,6 +15,8 @@ extern "C" {
 
 extern UART_HandleTypeDef huart2;
 extern DMA_HandleTypeDef hdma_usart2_rx;
+extern osThreadId uart2_thread;
+extern int32_t uart2_signal;
```

Comment:
Do we really need to screw up code again with externs in cpp files? Why isn't there a header file that would describe the necessary data types and items.

Reply:
same externs already contained in this source, this is because HardwareSerial for TMC is hack, we need use some handles from kernel. Arduino HardwareSerial should be removed in future and we should control TMC directly via HAL functions.